pkg/gds/gds.go

```diff
@@ -412,6 +417,13 @@ func (s *GDS) Lookup(ctx context.Context, in *api.LookupRequest) (out *api.Looku
 		return nil, status.Error(codes.NotFound, "no VASP record available")
 	}
 
+	// send lookup request activity to network activity handler
+	if vasp != nil {
+		go activity.Lookup().VASP(uuid.MustParse(vasp.Id)).Add()
```

Comment:
I believe we should always have a non-nil vasp here, if it was nil then line 416 would have panicked. Also if you want to change the `VASP()` method to accept a string rather than a uuid that would be fine with me. That way we wouldn't have to worry about the panic from `MustParse` here, although it means we have to be more careful about making sure what we pass in is actually a VASP ID.